### PR TITLE
Improve ChatInterface notebook

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
Builds on top of #5333. So review that one first.

FYI. @ahuang11 . Started reviewing the `ChatInterface` via this PR.

### CODE REVIEW

File: panel/widgets/chat.py [hunk 1]
```diff
@@ -30,6 +31,14 @@
 from .input import FileInput, TextInput
 
 
+def dataframe_renderer(value):
+    layout = Column(f"{len(value)} rows was uploaded")
+    if len(value)>1000:
+        value = value.sample(1000)
+        layout.insert(-1, "The data was downsampled to 1000 rows")
+    # Todo: We need to figure out how to tell users to add "perspective" to pn.extension
+    return layout.insert(-1, Perspective(object=value, sizing_mode="stretch_width", max_height=500))
```
Comment (ahuang11) [inline panel/widgets/chat.py:40]: Can you explain why Perspective over pn.pane.DataFrame?

Comment (MarcSkovMadsen) [reply panel/widgets/chat.py:40]: Pn.pane.DataFrame is so difficult when you and the user don't know the data uploaded.

It cannot really handle a lot of data nicely. Adding scroll bars and a max height would help.

The user cannot really explore the data. Filter, sort, pivot and Chart.

Comment (ahuang11) [reply panel/widgets/chat.py:40]: How big is the `perspective` extension?

**Comment by ahuang11** [comment]
Working on this PR.

**Comment by ahuang11** [comment]
Will merge now; feel free to review the main branch after

**Pull request merged**